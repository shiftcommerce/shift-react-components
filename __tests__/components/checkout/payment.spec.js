// Libraries
import React from 'react'

// Components
import Payment from '../../../src/components/checkout/payment'
import PaymentHeader from '../../../src/components/checkout/payment-header'
import StripePayment from '../../../src/components/checkout/stripe-payment'

test('renders the payment header and stripe payment component', () => {
  // Arrange
  const dummyClassName = 'dummy-classname'
  const headerTitle = 'Payment Details'

  // Act
  const wrapper = shallow(<Payment cart={{}} order={{}} title={headerTitle} className={dummyClassName} />)

  // Arrange
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('div').first()).toHaveClassName(dummyClassName)
  expect(wrapper.find(PaymentHeader).length).toEqual(1)
  expect(wrapper.find(StripePayment).length).toEqual(1)
})
