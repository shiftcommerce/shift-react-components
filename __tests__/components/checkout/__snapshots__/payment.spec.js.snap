// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the payment header and stripe payment component 1`] = `
<div
  aria-label="Payment method"
  className="dummy-classname o-form c-payment"
>
  <PaymentHeader />
  <div
    className="c-payment__section"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <StripePayment />
  </div>
</div>
`;
